### PR TITLE
Add new Braze newsletter banner

### PR DESCRIPTION
**Pull request by KaliedaRik**
## What does this change?
This PR:
+ Adds a small backend change to get Frontend to send a function to Braze which Braze banners can then use for single-click newsletter signup - this replicates existing functionality for Braze epic messages.
+ Bumps the version of the `braze-components` dependency to latest

## Does this change need to be reproduced in dotcom-rendering ?

- [ ] No
- [x] Yes - see PR https://github.com/guardian/dotcom-rendering/pull/7368

## Screenshots
Nothing interesting to show here

### Tested
- [x] Locally
- [x] On CODE

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/commercial/braze/brazeBanner.tsx [hunk 1]
```diff
@@ -65,6 +65,8 @@ const renderBanner = (
 			document.body.appendChild(container);
 
 			const Component = brazeModule.BrazeBannerComponent;
+			const idApiUrl: string = config.get('page.idApiUrl') ?? '';
```
Comment (tjmw) [inline static/src/javascripts/projects/common/modules/commercial/braze/brazeBanner.tsx:68]: If we're unable to get the `idApiUrl`, I wonder if we should fail in `subscribeToNewsletter with a more explicit error? Currently we'll attempt to call fetch with a path but no domain. I think in that case `fetch` assumes the current hostname, so `https://www.theguardian.com/users/me/newsletters`, which will fail and therefore should cause the newsletter banner to display its error message. But could be confusing when attempting to debug. What do you think?

Comment (KaliedaRik) [reply static/src/javascripts/projects/common/modules/commercial/braze/brazeBanner.tsx:68]: Hi Tom. I went searching for `page.idApiUrl` through the code base. It's not a common thing - the only other module using it is `identity` - for example see [identity/api.ts](https://github.com/guardian/frontend/blob/main/static/src/javascripts/projects/common/modules/identity/api.ts#L96) where it gets picked up with an alternate nonce root of `/ID_API_ROOT_URL_NOT_FOUND`.

Even in that file, I only found one example of any attempt to action an error - see line [196](https://github.com/guardian/frontend/blob/main/static/src/javascripts/projects/common/modules/identity/api.ts#L196) - which makes me think that if the page fails to load the config data which includes `page.idApiUrl` then the page has much bigger issues than our attempt to create a one-click newsletter signup function.

What sort of error message did you have in mind? What sort of tracking data should surround it? And where would we be best placed to handle the fail /display the message - in Frontend, or the Banner component itself?

Comment (tjmw) [reply static/src/javascripts/projects/common/modules/commercial/braze/brazeBanner.tsx:68]: > which makes me think that if the page fails to load the config data which includes page.idApiUrl then the page has much bigger issues than our attempt to create a one-click newsletter signup function.

Hey yep this is a good point, if this isn't available then perhaps things have gone pretty wrong on the page already so maybe it's not something we need to worry about affecting the Braze banner too much.

> What sort of error message did you have in mind? What sort of tracking data should surround it? And where would we be best placed to handle the fail /display the message - in Frontend, or the Banner component itself?

All good questions! I was imagining that perhaps in the subscribeToNewsletter callback, if the idapi base URL isn't available we could log an error to Sentry (so at the point a user clicks on subscribe)?

> A second thought: would it be worthwhile adding a check somewhere which, if the code failed to pickup the root url, would then suppress any attempt to load a newsletter banner in the first place - why display a newsletter signup if the signup is guaranteed to fail for that user? How do newsletter epics handle this situation?

More good points! Right now I don't think any code on the platforms is aware of specific React components, which is nice as it reduces coupling. In order to do this I guess the platform would need to be aware of the newsletter banner. I think on DCR the idapi base URL is one of the required fields for rendering an epic, so if it's not available no epic would be shown.

Overall perhaps this is something to mull over and address in a future PR. The component itself is setup to handle any errors in the callback already by showing an error message I think.

File: package.json [hunk 2]
```diff
@@ -17,7 +17,7 @@
 		"@guardian/ab-core": "^4.0.0",
 		"@guardian/atom-renderer": "1.2.2",
 		"@guardian/automat-modules": "^0.3.8",
-		"@guardian/braze-components": "^11.0.0",
+		"@guardian/braze-components": "^10.0.1-0",
```
Comment (tjmw) [inline package.json:20]: Just a reminder that I think we said after merging the braze-components PR we'd do an v11.1.0 release so we'd update to that version here.

Comment (KaliedaRik) [reply package.json:20]: yes

**Comment by KaliedaRik** [comment]
The current typescript error is expected and should resolve once we update the braze-component dependency to match latest braze-component version following the merge of this BC branch into production https://github.com/guardian/braze-components/pull/394

**Review by tomrf1 — APPROVED** [review]
👍

**Pull request merged**